Source/DafnyLanguageServer/Workspace/DocumentDatabase.cs

```diff
@@ -166,11 +171,17 @@ IRelocator relocator
           SymbolTable = relocator.RelocateSymbols(oldDocument.SymbolTable, documentChange, CancellationToken.None),
           SerializedCounterExamples = null,
           LoadCanceled = true,
-          OldVerificationDiagnostics = migratedVerificationDiagnotics
+          OldVerificationDiagnostics = migratedVerificationDiagnotics,
+          VerificationTree = migratedVerificationTree
         };
       }
     }
 
+    private DafnyDocument WithRealtimeDiagnosticsPublished(DafnyDocument document) {
```

Comment:
I initially thought this method performs a mutating operation on `DafnyDocument`, since most `With` methods return a value that's different from the input value. I'd prefer inlining this method.